### PR TITLE
Add JavaDoc and test wrt custom `JsonDeserializer`s `null` handling 

### PR DESCRIPTION
**Pull request by JooHyukKim**
Adds JavaDoc and test regarding https://github.com/FasterXML/jackson-databind/issues/4225

### CODE REVIEW

File: src/test/java/com/fasterxml/jackson/databind/deser/CustomDeserializers4225NullCacheTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.testutil.DatabindTestUtil;
+import org.junit.jupiter.api.Test;
```
Comment (cowtowncoder) [inline src/test/java/com/fasterxml/jackson/databind/deser/CustomDeserializers4225NullCacheTest.java:11]: I'll fix these, import order is wrong.

**Review by cowtowncoder — APPROVED** [review]
